### PR TITLE
Bonus guide: Lightning Terminal => Pool & Loop web UI, Faraday CLI, Web Terminal connection (replaces #861)

### PR DESCRIPTION
**Pull request by VajraOfIndra**
This PR replaces #861, which was closed because of merging conflicts.
The guide is now a bonus guide and has been shortened a bit.

#### What

This is a proposal to add Lightning Terminal, with Loop, Pool and Faraday integrated, a Loop & Pool web UI and connection to the Web Terminal.

### Why

* Lightning Terminal can be run on top of LND as one single daemon that integrates the Loop, Pool and Faraday daemons.

  * **Loop** allows to loop sats in/out of channels (= submarine swaps) via Lightning Labs Loop node.
    * This tool can be very useful to end-users and merchants to quickly and easily manage their channel balances (merchants need to regualrly move sats to remote and onchain, while end-users might need to move back sats to the local side after spending an entire channel). The tool can also be used by routing node operators.

  * **Pool** is a channel marketplace. 
    * This tool can be very useful for merchants to get inbound liqudiity and for routers to earn an income.

  * **Faraday** is an accounting tool for advanced users (CLI-only, requires manipulation of JSON/CSV etc)

* It also provided a web UI than can be accessed inside a user LAN and offers a GUI for the Loop and Pool services.

* Running Lightning Terminal also allows to connect to the web Terminal via the recently developped Lighning Node Connect (LNC). This allows an operator to securely manage its node from anywhere.

* Running Lightning Terminal allows a user to loop in/out directely from the Ride The Lightning web UI

* The RaspiBolt is based on LND (for now) and Lightning Terminal is developped by the same people than LND. We should therefore expect continuous development of Lightning Terminal alongside LND and possibly added services in the future. Moreover, running LiT with LND remote avoids the bugs associated with running LiT with LND integrated (as was proposed before as an option in RaspiBolt v2).

TL;DR: Lightning Terminal gives access to very useful lightning services (i.e. Loop and Pool) that helps merchants, end-users and routers to manage their liquidity and also offers helpful dashboard that can be used locally or remotely. LiT should be maintained and improved upon in the future as it is developped by Lightning Labs, one of the main LN company.

#### How

* Created a new bonus guide and updated the lightning bonus index

#### Scope

- [ ] significant change to core configuration
- [x] independent bonus guide
- [ ] simple bug fix

#### Test & maintenance

Test this PR by following the guide and installing Lightning Terminal, visiting the web UIs and trying the first steps of the Pool and Loop examples.

### CODE REVIEW

File: guide/bonus/lightning/lightning-terminal.md [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+
+* With the "admin" user, create symlinks to the Lightning Terminal, Loop, Pool and Faraday directories
+
+  ```sh
+  $ ln -s /data/lit /home/admin/.lit
+  $ ln -s /data/loop /home/admin/.loop
+  $ ln -s /data/pool /home/admin/.pool
+  $ ln -s /data/faraday /home/admin/.faraday
+  ```
+
```
Comment (mktech-gh) [inline guide/bonus/lightning/lightning-terminal.md:132]:
```suggestion

* Display the links and check that they’re not shown in red (this would indicate an error)

$ ls -la
```

File: guide/bonus/lightning/lightning-terminal.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+---
+
+## Run Lightning Terminal
+
+### Manual start
+
+* Still with user “lit”, we first start Lightning Terminal manually to check if everything works fine.
+
+  ```sh
+  $ litd
```
Comment (mktech-gh) [inline guide/bonus/lightning/lightning-terminal.md:244]:
```suggestion
 $ sudo su - lit
  $ litd
```

File: guide/bonus/lightning/lightning-terminal.md [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+  ```
+
+* Test that Lightning Terminal is working by visiting the web UI
+  * Past the following URL in your browser: [https://raspibolt.local::8443/](https://raspibolt.local::8443/){:target="_blank"} (replace raspibolt.local by your node IP address if required)
```
Comment (mktech-gh) [inline guide/bonus/lightning/lightning-terminal.md:248]:
```suggestion
  * Past the following URL in your browser: [https://raspibolt.local:8443/](https://raspibolt.local:8443/){:target="_blank"} (replace raspibolt.local by your node IP address if required)
```

**Review by mktech-gh — CHANGES_REQUESTED** [review]
I installed Lightning Terminal on my RaspiBolt. It's up and running. Thanks

File: guide/bonus/lightning/lightning-terminal.md (hunk 1 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md:132]: Good idea, I added it.

File: guide/bonus/lightning/lightning-terminal.md (hunk 2 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md:244]: Instead of relogging, I deleted the previous command that was asking to exit the litd user session.

File: guide/bonus/lightning/lightning-terminal.md (hunk 3 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md]: Fixed

**Comment by VajraOfIndra** [comment]
> I installed Lightning Terminal on my RaspiBolt. It's up and running. Thanks

Thanks for the reviews, I amended the guide accordingly, PTAL.

File: guide/bonus/lightning/lightning-terminal.md (hunk 3 above)
Comment (mktech-gh) [reply guide/bonus/lightning/lightning-terminal.md]: I found a problem:

admin@raspibolt:~ $ lncli -version
lncli version 0.14.2-beta commit=lightning-terminal-v0.6.5-alpha




The script "Bonus guide: System overview " uses this command to determine the lnd version and now requests an update of the lnd:

.
.
up: 18:53       rx: 1.0GiB     2nd drive: none          available mem: 3132M
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
     .~~.   .~~.      ฿itcoin (mainnet)   up     Lightning (LND)     up
    '. \ ' ' / .'     v22.0.0                    **lightning-terminal-v0.6.5-alpha   Update!**
     .~ .~~~..~.      Sync    OK [#730857]       
.
.

Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md]: Right, I have the same issue.
But the issue is with the system overview script which should be corrected. Could you open an Issue to report this?

File: guide/bonus/lightning/lightning-terminal.md [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+
+* Close the browser window, go back to the terminal and stop Lightning Terminal by pressing Ctrl+C. Then exit the "lit" user.
+
+  ```sh
+  $ exit
+  ```
+
+### Autostart on boot
+
+Now we’ll make sure Lightning Terminal starts as a service on the Raspberry Pi so it’s always running. In order to do that, we create a systemd unit that starts the service on boot directly after LND.
```
Comment (RooSoft) [inline guide/bonus/lightning/lightning-terminal.md:261]: It's technically called a `service` if I'm not mistaken

File: guide/bonus/lightning/lightning-terminal.md [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+  
+* Check the live logging activity. Press Ctrl+C to exit.
+  
+  ```sh
+  $ sudo journalctl -f -u litd
+  ```
+
+### Aliases
+
+For now, softwares packaged in Lightning Terminal are all listening to the same port 8443. This is not the default behavior set in the code of these sofware so you must always indicate the RPC port as well as the TLS certificate of Lightning Terminal when using them, using flags (e.g., `pool --rpcserver=localhost:8443 --tlscertpath=~/.lit/tls.cert accounts list`, do not try it now as Lightning Terminal is not running yet!).
```
Comment (RooSoft) [inline guide/bonus/lightning/lightning-terminal.md:338]: Software alway is written in singular format: `software`

File: guide/bonus/lightning/lightning-terminal.md [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+
+* To use Loop, Pool or Faraday from the CLI, use the alias followed by the desired command (e.g., `litpool accounts list`, much simpler!)### Aliases
+
+### Settings
+
+* Log in to your Lightning Terminal by pasting [https://raspibolt.local:8443/](https://raspibolt.local:8443/){:target="_blank"} in your web browser (replace rapsibolt.local by your node IP address if needed).
+
+* In the left menu, click on "Settings"
+
+* Click on "Channel balance mode" and select the case that best corresponds to your usage of your LN node
```
Comment (RooSoft) [inline guide/bonus/lightning/lightning-terminal.md:373]: Click on "Channel balance mode", make a selection according to your LN node type

* Merchant: "Receiving"
* Casual end user: "Sending"
* Routing node: "Routing"

File: guide/bonus/lightning/lightning-terminal.md [hunk 7]
```diff
@@ -0,0 +1,549 @@
...
+  
+  alias litfaraday="frcli --rpcserver=localhost:8443 --tlscertpath=~/.lit/tls.cert"
+  alias litloop="loop --rpcserver=localhost:8443 --tlscertpath=~/.lit/tls.cert"
+  alias litpool="pool --rpcserver=localhost:8443 --tlscertpath=~/.lit/tls.cert" 
+  ```
+
+* Activate the aliases
+
+  ```sh
+  $ source .bashrc
```
Comment (RooSoft) [inline guide/bonus/lightning/lightning-terminal.md:362]: Is sourcing `~/.bashrc` really refreshing the `~./bash_aliases` file?

Just unsure... never heard of the latter.

File: guide/bonus/lightning/lightning-terminal.md [hunk 8]
```diff
@@ -0,0 +1,549 @@
...
+* In the left menu, click on "Settings"
+
+* Click on "Channel balance mode" and select the case that best corresponds to your usage of your LN node
+  * Choose "Receiving" if you plan to mostly receive LN payments (e.g. if you're a merchant)
+  * Choose "Sending" if you plan to mostly send LN payments (e.g. if you're a casual end-user)
+  * Choose "Routing" if your main goal is to route payments (e.g. if you're a LN router)
+
+* Go back to "Settings"
+
+* Click on "Bitcoin Transaction URL" and replace the default URL (`https://mempool.space/tx/{txid}`) by your own explorer (`https://raspibolt.local:4000/tx/{txid}`, replace raspibolt.local by your node IP if necessary). This will preserve your privacy when looking up transactions from the Terminal.
```
Comment (RooSoft) [inline guide/bonus/lightning/lightning-terminal.md:380]: A lot of parenthesis in the next couple of lines... I'd rephrase that...

**Review by RooSoft — CHANGES_REQUESTED** [review]
Great PR as always....

Just a couple of minor tweaks here and there...

File: guide/bonus/lightning/lightning-terminal.md (hunk 4 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md:261]: Afaik, technically, we create a unit file that will manage the service.
E.g.: https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files
image

I updated to "we create a systemd service unit file"

File: guide/bonus/lightning/lightning-terminal.md (hunk 6 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md:373]: Much more concise, I like that, fixed!

File: guide/bonus/lightning/lightning-terminal.md (hunk 7 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md:362]: Because `~/.bashrc`  contains:
```
# Alias definitions.
# You may want to put all your additions into a separate file like
# ~/.bash_aliases, instead of adding them here directly.
# See /usr/share/doc/bash-doc/examples in the bash-doc package.

if [ -f ~/.bash_aliases ]; then
    . ~/.bash_aliases
fi
```

wouldn't sourcing it enable the aliases..?

I found a tutorial sourcing `.bashrc` and one sourcing `.bash_aliases `so... not too sure which one is best to source

File: guide/bonus/lightning/lightning-terminal.md (hunk 8 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md]: I removed them all... :cry:    
:stuck_out_tongue_winking_eye:

**Comment by VajraOfIndra** [comment]
> Great PR as always....

:heart: 

> Just a couple of minor tweaks here and there...

Thanks @RooSoft for the review! See my replies and updates above PTAL

File: guide/bonus/lightning/lightning-terminal.md (hunk 7 above)
Comment (RooSoft) [reply guide/bonus/lightning/lightning-terminal.md:362]: You are right, nothing to change here... since `~/.bashrc` sources `~/.bash_aliases`, sourcing either of them works.

File: guide/bonus/lightning/lightning-terminal.md (hunk 4 above)
Comment (RooSoft) [reply guide/bonus/lightning/lightning-terminal.md]: Ok, `unit` is a higher level concept. A `service` a type of `unit` and thus is more specific.

I still stand by the fact that what you are creating is a `service`. Never heard of the term `unit` before. Thanks for the learning opportunity.

File: guide/bonus/lightning/lightning-terminal.md (hunk 8 above)
Comment (RooSoft) [reply guide/bonus/lightning/lightning-terminal.md]: Parenthesis are evil.

**Review by RooSoft — APPROVED** [review]
Approved it, but I'd still prefer a systemd service not being called a unit.

**Comment by VajraOfIndra** [comment]
> Approved it, but I'd still prefer a systemd service not being called a unit.

Thanks. I removed the term "unit" and used "service" only.

File: guide/bonus/lightning/lightning-terminal.md (hunk 4 above)
Comment (VajraOfIndra) [reply guide/bonus/lightning/lightning-terminal.md]: Fixed, removed unit andusing service only now.
